src/main/scala/dx/cwl/CwlValue.scala

```
@@ -975,7 +982,8 @@ object FileValue {
                 throw new Exception(s"invalid secondaryFiles value ${other}")
             },
             fields.get("format").map(PathValue.unwrapString),
-            fields.get("contents").map(PathValue.unwrapString)
+            fields.get("contents").map(PathValue.unwrapString),
+            fields.get("metadata").map(_.prettyPrint)
```

What if `None` returned from `get` here?

---

`fields.get("metadata").map(_.prettyPrint)` will return `None` if `metadata` field does not exist.

---

I mean I get _what_ is returned. The question is if `_` is `None` then `None.prettyPrint` should throw an exception at runtime

---

> Expression type:
> 
> Option[String]
> Symbol signature:
> 
> final def map[B](f: JsValue => B): Option[B]
> Returns a [scala.Some](vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/workbench.html) containing the result of applying f to this [scala.Option](vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/workbench.html)'s value if this [scala.Option](vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/workbench.html) is nonempty. Otherwise return None.
> 
> This is equivalent to:
> 
> option match {
>   case Some(x) => Some(f(x))
>   case None    => None
> }

---

But you mean it would still do `None.prettyprint`? @Gvaihir

---

I thought so.